code/modules/mob/inventory.dm

```diff
@@ -299,6 +299,12 @@
 	if(equipped)
 		if(W.flags_armor_protection)
 			add_limb_armor(W)
+		if(isclothing(W))
+			var/obj/item/clothing/equipped_clothing = W
+			if(equipped_clothing.tint)
+				adjust_tinttotal(equipped_clothing.tint)
```

Comment:
This is the place where it's bugging. `equipped` needs to differentiate between equipping on mob and equipping insidea storage container.